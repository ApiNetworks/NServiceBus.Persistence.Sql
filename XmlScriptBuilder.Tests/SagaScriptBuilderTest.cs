using System.IO;
using System.Text;
using ApprovalTests;
using NServiceBus.Persistence.SqlServerXml;
using NUnit.Framework;

[TestFixture]
public class SagaScriptBuilderTest
{
    [Test]
    public void BuildCreateScript()
    {
        var saga = new SagaDefinition
        {
            Name = "theSaga",
            CorrelationMember = "Property1",
        };

        var builder = new StringBuilder();
        using (var writer = new StringWriter(builder))
        {
            SagaScriptBuilder.BuildCreateScript(saga, writer);
        }
        var script = builder.ToString();

        SqlValidator.Validate(script);
        Approvals.Verify(script);
    }

    [Test]
    public void BuildDropScript()
    {
        var builder = new StringBuilder();
        using (var writer = new StringWriter(builder))
        {
            SagaScriptBuilder.BuildDropScript("theSaga", writer);
        }
        var script = builder.ToString();
        SqlValidator.Validate(script);
        Approvals.Verify(script);
    }
}
