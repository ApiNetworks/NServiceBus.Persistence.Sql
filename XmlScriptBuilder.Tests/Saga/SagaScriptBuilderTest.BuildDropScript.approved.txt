
declare @tableName nvarchar(max) = '[' + @schema + '].[' + @endpointName + '.theSaga]';

IF EXISTS
(
    SELECT *
    FROM sys.objects
    WHERE
        object_id = OBJECT_ID(@tableName)
        AND type in (N'U')
)
BEGIN
    DECLARE @createTable nvarchar(max);
    SET @createTable = N'DROP TABLE ' + @tableName + '';
    exec(@createTable);
END
